### ISSUE TITLE
bad archive

### ISSUE THREAD

**Issue by coot** [cachix/cachix#147]
Is there a way to re-upload or remove a bad archive, I am getting this error while building using `cachix`:
```
copying path '/nix/store/k816vh2ia2ar3lbfrwp9xaf6dszcvvzy-cryptonite-0.25' from 'https://ouroboros-network.cachix.org'...
bad archive: input doesn't look like a Nix archive
copying path '/nix/store/p1nh69clk8y49alpkh4dy2qdj05r6j1j-old-locale-1.0.0.7' from 'https://cache.nixos.org'...
cannot build derivation '/nix/store/0yimac9m3ch87qdbim3y3wjhzzx5sgi4-ouroboros-network-0.1.0.0.drv': 1 dependencies couldn't be built
error: build of '/nix/store/0yimac9m3ch87qdbim3y3wjhzzx5sgi4-ouroboros-network-0.1.0.0.drv' failed

travis_time:end:0c422119:start=1541447696419846326,finish=1541447757856048335,duration=61436202009
[0K
[31;1mThe command "nix-build --arg dev true -A ouroboros-network -o result --argstr compiler $COMPILER" exited with 1.[0m

Done. Your build exited with 1.
```

**Comment by domenkozar** [comment]
@coot removed from your cache.

**Comment by coot** [comment]
thanks @domenkozar

**Comment by domenkozar** [comment]
I haven't verified yet, but I think the problem is that stream end prematurely and conduit just finishes without an error. Not sure how to address this yet, but I'll prioritize it.

**Comment by dtzWill** [comment]
I think when I encountered this it's possible (likely even) I killed the client mid-upload.  Story fits, for whatever it's worth :).

**Comment by domenkozar** [comment]
@coot could you run `cachix --version` on the machine this nar was uploaded from?

**Comment by domenkozar** [comment]
I fixed one potential bug in 0.1.2, but it could be there's another source. Going to wait on another report that confirms it's reproducible with 0.1.2. Thanks!

**Comment by domenkozar** [comment]
Going to close this and keep an eye on more reports.
